src/cloud-resources/src/crd/console.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+
+use k8s_openapi::{
+    api::core::v1::ResourceRequirements, apimachinery::pkg::apis::meta::v1::Condition,
+};
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::crd::{ManagedResource, MaterializeCertSpec, new_resource_id};
+use mz_server_core::listeners::AuthenticatorKind;
+
+pub mod v1alpha1 {
+    use super::*;
+
+    #[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize, JsonSchema)]
+    #[serde(rename_all = "camelCase")]
+    pub struct BalancerdRef {
+        // The service name for the balancerd service to connect to
+        pub service_name: String,
+        // The namespace the balancerd service runs in
+        pub namespace: String,
+        // The configuration for generating an x509 certificate using cert-manager for balancerd
+        // to present to incoming connections.
+        // The dns_names and issuer_ref fields are required.
+        pub external_certificate_spec: Option<MaterializeCertSpec>,
+        /// How to authenticate with Materialize.
+        #[serde(default)]
+        pub authenticator_kind: AuthenticatorKind,
```

Comment:
Nitpick, should this be in the BalancerdRef, or should it just be in the ConsoleSpec?

Reply:
my understanding is that this is only used for determining how console talks to balancerd? or am i missing something